example/transform/huggingface_pdf_source_irs_QA.ipynb

```diff
@@ -0,0 +1,1323 @@
+   "source": [
+    "#### Create extract_config"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "9ddba5ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "extract_config = ExtractPDFConfig(\n",
+    "    model_config=NougatModelConfig(\n",
+    "        model_name=\"0.1.0-small\",\n",
+    "        batch_size=1,  # When batch_size>1, nougat will run on CUDA, otherwise it will run on CPU\n",
+    "    ),\n",
+    "    splitter=MARKDOWN_HEADER_SPLITTER,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d7b5a712",
+   "metadata": {},
+   "source": [
+    "#### Create extract_client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "cae041e2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/conda/envs/uniflow/lib/python3.10/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3526.)\n",
+      "  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]\n"
+     ]
+    }
+   ],
+   "source": [
+    "extract_client = ExtractClient(extract_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd71259d",
+   "metadata": {},
+   "source": [
+    "#### Run extract_client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "8ab2ec60",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "  0%|          | 0/1 [00:00<?, ?it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "INFO: likely hallucinated title at the end of the page: ## Costs You Can Deduct or Capitalize Page 27\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 1/1 [08:53<00:00, 533.49s/it]\n"
+     ]
+    }
+   ],
+   "source": [
+    "extract_output = extract_client.run(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "efc1aa41",
+   "metadata": {},
+   "source": [
+    "#### Prepare sample prompts"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cbc9dccf",
+   "metadata": {},
+   "source": [
+    "First, we need to demonstrate sample prompts for LLM. We do this by giving a sample instruction to the PromptTemplate class"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "45f0b5e8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_instruction = \"\"\"Assume you are an expert on tax, please generate as many question as possible based on the context. \n",
+    "Make sure those questions can cover any question people can think of by reading the context.\"\"\"\n",
+    "\n",
+    "guided_prompt = PromptTemplate(instruction=sample_instruction)"
+   ]
```

Comment:
You should consider use few shot prompt to stabilize the output. Check the example [here](https://github.com/CambioML/uniflow/blob/main/example/transform/huggingface_model.ipynb) for the `Prepare sample prompts` section. With this, your output should be more stable, so you will not need the complicated postprocessing logic.